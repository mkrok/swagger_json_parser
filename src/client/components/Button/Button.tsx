import * as React from 'react';
import styled from 'styled-components';

const ButtonStyled = styled.button`
  background-color: #800;
  color: #ccc;
  margin: 1em;
  padding: 0.25em 1em;
  border: 1px solid #aaa;
  font-size: 0.8em;
  &:hover {
    background-color: #600;
  }
`;

const Button = ({ onClick, text }) => <ButtonStyled onClick={onClick}>{text}</ButtonStyled>;

export default Button;
