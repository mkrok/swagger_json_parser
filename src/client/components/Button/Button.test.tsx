import * as React from 'react';
import renderer from 'react-test-renderer';
import Button from './';

describe('Button', () => {
  it('should match snapshot', () => {
    const buttonProps = {
      text: 'Hello',
      onClick: () => {},
    };
    const component = renderer.create(<Button {...buttonProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
