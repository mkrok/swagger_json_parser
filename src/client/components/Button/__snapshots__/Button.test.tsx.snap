// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="sc-beqWaB bZIEFo"
  onClick={[Function]}
>
  Hello
</button>
`;
